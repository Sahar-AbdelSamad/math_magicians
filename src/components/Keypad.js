import { v4 as uuidv4 } from 'uuid';
import PropTypes from 'prop-types';

const Keypad = ({ click }) => {
  const btn = ['AC', '+/-', '%', 'รท', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', ' ='];
  return (
    <div className="buttons">
      {btn.map((value) => (
        <button type="button" key={uuidv4()} onClick={click}>
          {value}
        </button>
      ))}
    </div>
  );
};

Keypad.propTypes = {
  click: PropTypes.func.isRequired,
};
export default Keypad;
