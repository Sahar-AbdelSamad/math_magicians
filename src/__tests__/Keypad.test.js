import renderer from 'react-test-renderer';
import Keypad from '../app/components/Keypad';

it('renders correctly', () => {
  const tree = renderer
    .create(<Keypad />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
