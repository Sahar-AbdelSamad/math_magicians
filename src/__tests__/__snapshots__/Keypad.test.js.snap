// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="buttons"
>
  <button
    type="button"
  >
    AC
  </button>
  <button
    type="button"
  >
    +/-
  </button>
  <button
    type="button"
  >
    %
  </button>
  <button
    type="button"
  >
    รท
  </button>
  <button
    type="button"
  >
    7
  </button>
  <button
    type="button"
  >
    8
  </button>
  <button
    type="button"
  >
    9
  </button>
  <button
    type="button"
  >
    x
  </button>
  <button
    type="button"
  >
    4
  </button>
  <button
    type="button"
  >
    5
  </button>
  <button
    type="button"
  >
    6
  </button>
  <button
    type="button"
  >
    -
  </button>
  <button
    type="button"
  >
    1
  </button>
  <button
    type="button"
  >
    2
  </button>
  <button
    type="button"
  >
    3
  </button>
  <button
    type="button"
  >
    +
  </button>
  <button
    type="button"
  >
    0
  </button>
  <button
    type="button"
  >
    .
  </button>
  <button
    type="button"
  >
     =
  </button>
</div>
`;
